app/src/main/kotlin/cz/covid19cz/app/service/CovidService.kt

```diff
@@ -196,4 +209,63 @@ class CovidService : Service() {
             true
         }
     }
+
+    /**
+     * Inits regular check to ensure we're reading data when the screen is off.
+     */
+    private fun initScreenOffScanCheck() {
+        Handler(Looper.getMainLooper()).post {
+            screenSwitchedOffTime = null
+
+            // scan for 30m, each 15s
+            screenOfDetectionTimer =
+                object : CountDownTimer(30 * 60 * 1000, 15000) {
+                    override fun onTick(millisUntilFinished: Long) {
+                        btUtils.lastScanResultTime.value?.let { lastResultTime ->
+                            ensureReceivingResults( // pass local R-O copies
+                                lastResultTime,
+                                screenSwitchedOffTime
+                            )
+                        }
+                    }
+
+                    override fun onFinish() {}
+                }.start()
+        }
+    }
+
+    /**
+     * Checks if the screen is off and if so, ensures, that we're still receiving BT data.
+     * If the receiving is stopped, it's restarted.
+     */
+    private fun ensureReceivingResults(lastResultTime: Long, switchedOffSince: Long?) {
+        if (!isScreenOn()) {
+            if (switchedOffSince != null) {
+                if (lastResultTime <= switchedOffSince + 1000) { // 1s tolerance
+                    Log.d("Device does NOT support scanning on background with empty filter, restarting")
+                    useScanFilter = true
+                    btUtils.stopScanning()
+                    btUtils.startScanning(useScanFilter = true)
+                    screenOfDetectionTimer?.cancel() // cancel this check, it's not needed
```

Comment:
A z testovani teda plyne, ze kdyz to aspon neco zmeri, tak tam tenhle problem neni..Nechtelo by to zvetsit to okno tolerance?

Reply:
Jako stalo se mi, že přišel update "těsně po" zhasnutí, ale bylo to tak těsný, že spíš současný - na základě toho jsem tam udělal tu vteřinu. Ale klidně jí zvětším, jestli myslíš.